### ISSUE TITLE
AsyncGRPOTrainer: loss normalization across gradient accumulation and complete-rollout admission

### ISSUE THREAD

**Issue by adithya-s-k** [huggingface/trl#7206]
When using TRL’s experimental `AsyncGRPOTrainer` for multi-turn agent training, one rollout can produce multiple `RolloutSample` sequences—for example, when conversation history changes and the token-prefix chain forks. Each sequence carries the rollout’s advantage, while historical context remains masked.

Two behaviors in this training path are worth reviewing:

**1. Loss normalization across gradient accumulation**

`compute_loss()` normalizes by `global_n_tokens` for the current microbatch, then divides by the gradient accumulation count. Although the token count covers distributed ranks, independently normalizing each microbatch makes effective token weighting depend on packing.

For example, within one optimizer update:

| Microbatch | Supervised tokens | Mean token loss |
|---|---:|---:|
| A | 100 | 2.0 |
| B | 900 | 0.5 |

Averaging microbatch losses gives **1.25**; normalizing across all supervised tokens gives **0.65**.

For a token-mean objective, the denominator should cover the entire accumulation window and all ranks, with distributed gradient averaging handled consistently.

**2. Rollout identity and staleness checks**

`_score_group()` emits separate samples for a rollout’s sequences, and `RolloutQueueDataset` checks staleness independently for each sample:

```text
Rollout R produces S1, S2, S3
→ S1 is consumed
→ training advances the policy version
→ S2 and S3 exceed max_staleness and are dropped
```

This permits partial rollout consumption. Retaining `rollout_id` and sequence membership would allow complete-rollout admission before `TokenBudgetBatcher` packs the sequences.

**Expected behavior:** repacking identical supervised tokens and causal contexts should preserve their loss weighting, and every sequence from an admitted rollout should be accounted for. Useful regression cases include unequal supervised-token counts across microbatches/ranks and policy-version changes between reads of the same rollout.
